### PR TITLE
feat: [CDS-58893]: Deprecate Kustomize and use Kubectl instead

### PR DESCRIPTION
**Pull request by alebuh**
## Describe your changes
Remove Kustomize installation from Immutable delegate docker images
Add logic to use Kubectl if Kustomize is not installed on the delegate
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48815)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger uts, publish-delegate

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger publish-delegate

**Comment by abhinavsingh1196** [comment]
@harness/del-team to review the changes for removal of kustomize binary from immutable delegate

**Comment by alebuh** [comment]
trigger uts

File: 960-api-services/src/test/java/io/harness/k8s/kubectl/KustomizeCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Map;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@OwnedBy(HarnessTeam.CDP)
+public class KustomizeCommandTest extends CategoryTest {
+  @Test
+  @Owner(developers = BUHA)
+  @Category(UnitTests.class)
+  public void smoke() {
```
Comment (TarunUba1) [inline 960-api-services/src/test/java/io/harness/k8s/kubectl/KustomizeCommandTest.java:29]: Smoke  ?💨

**Review by abosii-harness — APPROVED** [review]

**Review by namantalaycha10 — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Comment by markoharness** [comment]
*[`960-api-services/src/test/java/io/harness/kustomize/KubectlKustomizeClientImplTest.java` line 34 at r7](https://reviewable.io/reviews/harness/harness-core/48815#-NXQPjDo6b33EprF6Nwy:-NXQPjDo6b33EprF6Nwz:b7socqp) ([raw file](https://github.com/harness/harness-core/blob/52915d76bb125e96d739c845dd1615ea7e0e8d66/960-api-services/src/test/java/io/harness/kustomize/KubectlKustomizeClientImplTest.java#L34)):*
> ```Java
> import org.mockito.MockitoAnnotations;
> 
> public class KubectlKustomizeClientImplTest extends CategoryTest {
> ```

Missing the OwnedBy annotation

**Review by markoharness — COMMENTED** [review]
> Reviewed 5 of 13 files at r1, 2 of 3 files at r2, 4 of 9 files at r4, 1 of 1 files at r6, 7 of 7 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48815)* status: 19 file(s) to review. Waiting on team(s): del-team.

**Review by alebuh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48815)* status: 19 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`960-api-services/src/test/java/io/harness/kustomize/KubectlKustomizeClientImplTest.java` line 34 at r7](https://reviewable.io/reviews/harness/harness-core/48815#-NXQPjDo6b33EprF6Nwy:-NXThDaH-55brVh4HECt:b-6978hb) ([raw file](https://github.com/harness/harness-core/blob/52915d76bb125e96d739c845dd1615ea7e0e8d66/960-api-services/src/test/java/io/harness/kustomize/KubectlKustomizeClientImplTest.java#L34)):*
<details><summary><i>Previously, markoharness (Marko Lukic) wrote…</i></summary><blockquote>

Missing the OwnedBy annotation
</blockquote></details>

added now

**Comment by alebuh** [comment]
trigger uts

**Review by markoharness — DISMISSED** [review]
> Reviewed 1 of 1 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48815)* status: 19 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team,k8s-team.

**Comment by alebuh** [comment]
Closing this PR for now as it will not be merged before the middle of July due to the depreciation notice

**Comment by alebuh** [comment]
trigger uts

**Review by abosii-harness — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]

**Comment by alebuh** [comment]
trigger uts, review

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger uts

**Comment by alebuh** [comment]
trigger ut6

**Pull request merged**